packages/drizzle/src/updateOne.ts

```diff
@@ -74,23 +122,70 @@ export const updateOne: UpdateOne = async function updateOne(
     return null
   }
 
-  const result = await upsertRow({
-    id: idToUpdate,
+  if (!idToUpdate || shouldUseUpsertRow({ data, fields: collection.flattenedFields })) {
+    const result = await upsertRow({
+      id: idToUpdate,
+      adapter: this,
+      data,
+      db,
+      fields: collection.flattenedFields,
+      ignoreResult: returning === false,
+      joinQuery,
+      operation: 'update',
+      req,
+      select,
+      tableName,
+    })
+
+    if (returning === false) {
+      return null
+    }
+
+    return result
+  }
+
+  const { row } = transformForWrite({
     adapter: this,
     data,
-    db,
     fields: collection.flattenedFields,
-    ignoreResult: returning === false,
-    joinQuery,
-    operation: 'update',
-    req,
-    select,
     tableName,
   })
 
+  const drizzle = db as LibSQLDatabase
+  await drizzle
+    .update(this.tables[tableName])
+    .set(row)
+    .where(eq(this.tables[tableName].id, idToUpdate))
```

Comment:
In the future, if no id is provided, we should just use a Where query here instead of fetching `idToUpdate` beforehand. I'd add a todo for now